library/cmac.c

```diff
@@ -339,7 +340,7 @@ int mbedtls_cipher_cmac_finish( mbedtls_cipher_context_t *ctx,
     last_block = cmac_ctx->unprocessed_block;
 
     /* Calculate last block */
-    if( cmac_ctx->padding_flag )
+    if( cmac_ctx->padding_flag && cmac_ctx->unprocessed_len < block_size )
```

Comment:
Why is it necessary to have this extra condition here? I mean, if the padding flag is set, surely the unprocessed block length is definitely smaller that the cipher block size. If this is not the case, I would presume there is an inconsistency somewhere else.